src/impls.rs

```diff
@@ -262,15 +262,15 @@ impl<T: Config> Pallet<T> {
 	/// # Arguments
 	/// * `timestamp` - Data feed unique identifier.
 	/// * `max` - The maximum number of pending dispute votes to be sent.
-	pub(super) fn do_send_votes(timestamp: Timestamp, max: u8) -> DispatchResult {
+	pub(super) fn do_send_votes(timestamp: Timestamp, max: u8) -> Result<u32, DispatchError> {
 		let governance_contract = T::Governance::get();
 		const GAS_LIMIT: u64 = gas_limits::VOTE;
 		// Check for any pending votes to be sent to governance controller contract
 		let mut pending_votes: Vec<_> = <PendingVotes<T>>::iter()
 			.filter(|(_, (_, scheduled))| &timestamp >= scheduled)
 			.collect();
 		pending_votes.sort_by_key(|(_, (_, scheduled))| *scheduled);
-		for (dispute_id, (vote_round, _)) in pending_votes.into_iter().take(max.into()) {
+		for (dispute_id, (vote_round, _)) in pending_votes.clone().into_iter().take(max.into()) {
```

Comment:
Remove clone by capturing length into variable, or use .iter() instead of .into_iter()